### PR TITLE
Improved cron logging

### LINKED ISSUE #9474: Cron Statistics should reflect completion
Notifications were not being delivered, so first thing I checked if cron was running. The logs showed that cron was ran, but they failed to reflect that code halted due to fatal error and was never actually completed.

### PR DESCRIPTION
**Pull request by hypeJunction**
Improves cron logging and adds latest message to admin interface under cron statistics

Fixes #9474

### CODE REVIEW

**Comment by hypeJunction** [comment]
cron

**Comment by hypeJunction** [comment]
I almost wanted to add the timer here, but wasn't sure if it's of any use.

File: engine/lib/cron.php [hunk 1]
```diff
@@ -75,15 +72,22 @@ function _elgg_cron_run() {
 		$key = "cron_latest:$period:ts";
 		$ts = elgg_get_site_entity()->getPrivateSetting($key);
 		$deadline = $ts + $interval;
+
 		if ($now > $deadline) {
-			$old_stdout = elgg_trigger_plugin_hook('cron', $period, $params, $old_stdout);
```
Comment (hypeJunction) [inline engine/lib/cron.php]: Not sure how to deal with $return value here. If a plugin checks for return value before doing something, they are getting the wrong message that was added by another period... So, it's a bugfix, I suppose.

Comment (mrclay) [inline engine/lib/cron.php]: Agreed

**Comment by mrclay** [comment]
LGTM

**Comment by hypeJunction** [comment]
Hm, do we want to clear the message when cron starts? Or do we let site owners compare the timestamps?

**Comment by mrclay** [comment]
Are you talking about data passed to hook handlers? I'm not following.

**Comment by hypeJunction** [comment]
I am saying that in case of an unsuccessful cron job the message displayed will be from a previously succeeded cron.

**Comment by mrclay** [comment]
Can we just save the start time, then override it at the end?

**Comment by hypeJunction** [comment]
We can log something before calling a hook.

**Comment by hypeJunction** [comment]
@mrclay Now logs start time and then replaces it with output message. Also squashed the commits.

File: languages/en.php [hunk 2]
```diff
@@ -477,6 +477,9 @@
 	'admin:cron:period' => 'Cron period',
 	'admin:cron:friendly' => 'Last completed',
 	'admin:cron:date' => 'Date and time',
+	'admin:cron:msg' => 'Message',
+	'admin:cron:started' => 'Cron jobs for "%s" started at %s',
```
Comment (mrclay) [inline languages/en.php]: In the context of the table do we need "Cron jobs for period" in here? Why not "Started (time). May have failed or is still running."

Comment (mrclay) [inline languages/en.php]: Oh because multiple run. Ignore that.

**Comment by mrclay** [comment]
Looks good. I'd like to try it out.

**Pull request merged**